applications: nrf_desktop: battery_charger: use Devicetree (#8102)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/battery_charger.rst

By default, the module is disabled and the :ref:`CONFIG_DESKTOP_BATTERY_CHARGER_NONE <config_desktop_app_options>` option is selected.
Set the option :ref:`CONFIG_DESKTOP_BATTERY_CHARGER_DISCRETE <config_desktop_app_options>` to enable the module.

The module requires a node labeled ``battery_charger`` with ``battery-charger`` compatible set in Devicetree.

> Three properties are required: the CSO GPIO spec, the Enable GPIO spec and the CSO switching frequency.

Author: good point, let me add it

> Three properties are required: the CSO GPIO spec, the Enable GPIO spec and the CSO switching frequency.

Reviewer: @divipillai can tell for sure :)
We will need her approval here too.

> Three properties are required: the CSO GPIO spec, the Enable GPIO spec and the CSO switching frequency.

Reviewer 2: Agree and now its already added :).

> The module requires a node labeled ``battery_charger`` with ``battery-charger`` compatible set in Devicetree.
> Three properties are required: the CSO (Charge Status Output) GPIO spec, the Enable GPIO spec and the CSO switching frequency.
> Below you can find an example:

Reviewer 2: ```suggestion
The module requires a node labeled ``battery_charger`` with a ``battery-charger`` compatible set in Devicetree.
The charge status output (CSO) GPIO spec, the Enable GPIO spec, and the CSO switching frequency properties are also required.
See the following snippet for an example:
```
